Source/ModelGroup.cs

```diff
@@ -7,15 +7,23 @@ namespace AssetGenerator
 {
     internal abstract partial class ModelGroup
     {
-        public abstract ModelGroupName Name { get; }
-
-        public List<Property> CommonProperties = new List<Property>();
-        public List<Property> Properties = new List<Property>();
-        public List<Model> Models;
-        public List<Runtime.Image> UsedTextures = new List<Runtime.Image>();
-        public List<Runtime.Image> UsedFigures = new List<Runtime.Image>();
-        public int Id = -1;
-        public bool NoSampleImages = false;
+        internal abstract ModelGroupName Name { get; }
+        internal List<Model> Models { get; set; }
+        internal List<Property> CommonProperties { get; private set; }
+        internal List<Property> Properties { get; private set; }
+        internal List<Runtime.Image> UsedTextures { get; private set; }
+        internal List<Runtime.Image> UsedFigures { get; private set; }
+        internal int Id { get; set; }
```

Comment:
I forget what Id is used for. Does it need to be set publicly?

Reply:
Also, if we do keep it we should probably rename it to something like `Index`

Reply:
Yes, let's consider doing that, but probably not with this change.